synapse/util/task_scheduler.py

```diff
@@ -352,6 +352,9 @@ async def wrapper() -> None:
                 )
                 self._running_tasks.remove(task.id)
 
+            # Try launch a new task since we've finished with this one.
+            self._clock.call_later(1, self._launch_scheduled_tasks)
```

Comment:
Does `call_later` also take care of not launching if it is already running through a `looping_call` scheduling ? I can't really tell from Twisted doc.
If it's not this is probably not safe to do.

Reply:
Oh, for some reason I thought we already had the check that only one of these was running at a time. Will add.